### PR TITLE
The rest of Collection Tests

### PR DESCRIPTION
**Pull request by LaurenNguyen14**
Add some edge cases regarding naming scopes and collections, and test CollectionxDatabase

### CODE REVIEW

File: common/test/java/com/couchbase/lite/DbCollectionsTest.kt [hunk 1]
```diff
@@ -50,6 +74,31 @@ class DbCollectionsTest : BaseCollectionTest() {
...
+        scope = baseTestDb.getScope("3icro")
+        assertEquals(1, scope?.collectionCount)
+        assertNotNull(scope?.getCollection("chintz"))
+
+
+        scope = baseTestDb.getScope("-micro")
+        assertEquals(1, scope?.collectionCount)
+        assertNotNull(scope?.getCollection("chintz"))
+
+        //Illegal chars
```
Comment (bmeike) [inline common/test/java/com/couchbase/lite/DbCollectionsTest.kt:87]: Didn't you just test this, above?  What is is doing here?

File: common/test/java/com/couchbase/lite/DbCollectionsTest.kt [hunk 2]
```diff
@@ -50,6 +74,31 @@ class DbCollectionsTest : BaseCollectionTest() {
...
+        baseTestDb.createCollection("coll1", "scope1")
+        baseTestDb.createCollection("coll2", "Scope1")
+
+        val scope1 = baseTestDb.getScope("scope1")
+        val scope2 = baseTestDb.getScope("Scope1")
+
+        assertNotNull(scope1)
+        assertNotNull(scope2)
+        assertNotSame(scope1, scope2)
     }
```
Comment (bmeike) [inline common/test/java/com/couchbase/lite/DbCollectionsTest.kt:102]: You need to add to this test to verify that baseTestDb.getScope("scope1").equals(baseTestDb.getScope("scope1"))

File: common/test/java/com/couchbase/lite/DbCollectionsTest.kt [hunk 3]
```diff
@@ -333,4 +382,29 @@ class DbCollectionsTest : BaseCollectionTest() {
...
         recreateBastTestDb()
     }
-}
\ No newline at end of file
+
+    /**
+     * Collections and Cross Database instance
+     */
+
+    @Ignore("There is no collection on other db")
```
Comment (bmeike) [inline common/test/java/com/couchbase/lite/DbCollectionsTest.kt:390]: It seems that this should work.  If you can't remove the Ignore, can you say what is necessary to make it work?

**Review by bmeike — COMMENTED** [review]
Need to add to the scope name case sensitive test, and better explain the Ignore on the other db test.

**Review by bmeike — APPROVED** [review]
